libs/core/src/lib/generic-tag/generic-tag.component.ts

```diff
@@ -0,0 +1,100 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    Input,
+    OnChanges,
+    ViewEncapsulation,
+    OnInit
+} from '@angular/core';
+import { applyCssClass, CssClassBuilder, NullableObject, Nullable } from '@fundamental-ngx/cdk/utils';
+import { FD_GENERIC_TAG_COMPONENT } from './tokens';
+
+export type GenericTagType = 'error' | 'success' | 'warning' | 'information';
+
+@Component({
+    // eslint-disable-next-line @angular-eslint/component-selector
+    selector: '[fd-generic-tag]',
+    templateUrl: './generic-tag.component.html',
+    styleUrls: ['./generic-tag.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {
+            provide: FD_GENERIC_TAG_COMPONENT,
+            useExisting: GenericTagComponent
+        }
+    ],
+    host: {
+        '[attr.tabindex]': '0',
+        '[attr.aria-roledescription]': 'ariaRoleDescription',
+        role: 'button'
+    }
+})
+export class GenericTagComponent implements OnChanges, OnInit, CssClassBuilder {
+    /** User's custom classes */
+    @Input()
+    class: string;
+
+    /**
+     * The type of the Generic Tag.
+     * Can be one of the following: 'error' | 'success' | 'warning' | 'information'.
+     * For default Generic Tag omit this property.
+     */
+    @Input()
+    type: Nullable<GenericTagType>;
+
+    /**
+     * The KPI Name of the Generic tag. Standard text.
+     * Always use a meaningful title. Keep it simple and try to use no more than 3 words.
+     */
+    @Input()
+    name: Nullable<string>;
+
+    /**
+     * The KPI Value of the Generic tag.
+     * The value represents the numeric (key) attribute and its unit.
+     */
+    @Input()
+    value: Nullable<string>;
+
+    /**
+     *  Aria defines role description for the Generic Tag
+     */
+    @Input()
+    ariaRoleDescription: Nullable<string> = 'Generic Tag';
+
+    /** @hidden */
+    constructor(public readonly elementRef: ElementRef) {}
```

Comment:
You can use `inject` function instead of DI in constructor. This way it will be easier if some component would inherit this class as a base one.